nixos/tests/networking.nix

```diff
@@ -294,25 +294,25 @@ let
       };
       testScript = { ... }:
         ''
-          startAll;
+          start_all()
 
           # Wait for networking to come up
-          $client1->waitForUnit("network.target");
-          $client2->waitForUnit("network.target");
-          $router->waitForUnit("network.target");
-
-          # Test bridging
-          $client1->waitUntilSucceeds("ping -c 1 192.168.1.1");
-          $client1->waitUntilSucceeds("ping -c 1 192.168.1.2");
-          $client1->waitUntilSucceeds("ping -c 1 192.168.1.3");
-
-          $client2->waitUntilSucceeds("ping -c 1 192.168.1.1");
-          $client2->waitUntilSucceeds("ping -c 1 192.168.1.2");
-          $client2->waitUntilSucceeds("ping -c 1 192.168.1.3");
-
-          $router->waitUntilSucceeds("ping -c 1 192.168.1.1");
-          $router->waitUntilSucceeds("ping -c 1 192.168.1.2");
-          $router->waitUntilSucceeds("ping -c 1 192.168.1.3");
+          client1.wait_for_unit("network.target")
+          client2.wait_for_unit("network.target")
+          router.wait_for_unit("network.target")
+
+          with subtest("Test bridging"):
+              client1.wait_until_succeeds("ping -c 1 192.168.1.1")
+              client1.wait_until_succeeds("ping -c 1 192.168.1.2")
+              client1.wait_until_succeeds("ping -c 1 192.168.1.3")
+    
+              client2.wait_until_succeeds("ping -c 1 192.168.1.1")
+              client2.wait_until_succeeds("ping -c 1 192.168.1.2")
+              client2.wait_until_succeeds("ping -c 1 192.168.1.3")
+    
+              router.wait_until_succeeds("ping -c 1 192.168.1.1")
+              router.wait_until_succeeds("ping -c 1 192.168.1.2")
+              router.wait_until_succeeds("ping -c 1 192.168.1.3")
```

Comment:
how about:

```python
import itertools

def ping_until_succeeds(machine, ip):
    machine.wait_until_succeeds(f"ping -c 1 {ip}")


machines = [client1, client2, router]
ips = ["192.168.1.1", "192.168.1.2", "192.168.1.3" ]

with subtest("Test bridging"):
    for i in itertools.product(machines, ips):
        ping_until_succeeds(*i)
```

Reply:
In general I prefer simple, repetetive code in tests as opposed to more complex solutions.
At the moment you can read the test without knowing much about how python works.
Your example makes the code more compact / elegant, but also harder to understand.
I can refactor it if you want, but I don't think we should strive for idiomatic python in these tests.

Reply:
Don't see it as a hard requirement. I am fine with either solutions, i just like code to be compact which also reduces the risk for unseen typos